quinn-proto/src/connection.rs

```diff
@@ -837,6 +827,7 @@ impl Connection {
                     }
                     ConnectionError::TransportError { error_code } => State::closed(error_code),
                     ConnectionError::VersionMismatch => State::Draining,
+                    ConnectionError::StatelessReset => State::Drained,
```

Comment:
Instead of adding a new case, we should just get rid of the panic in the `Reset` case.

Reply:
Done.